traffic_ops/install/bin/postinstall.test.sh

```diff
@@ -0,0 +1,424 @@
+			"config_var": "cdn_name",
+			"hidden": false
+		},
+		{
+			"DNS sub-domain for which your CDN is authoritative": "cdn1.kabletown.net",
+			"config_var": "dns_subdomain",
+			"hidden": false
+		}
+	]
+}
+EOF
+
+"$MY_DIR/postinstall.py" --no-root --root-directory="$ROOT_DIR" --no-restart-to --no-database --ops-user="$(whoami)" --ops-group="$(whoami)" --automatic --cfile="$ROOT_DIR/defaults.json" --debug 2>"$ROOT_DIR/stderr" | tee "$ROOT_DIR/stdout"
+
+if [[ ! -z "$(grep 'ERROR' $ROOT_DIR/stderr)" ]]; then
+	echo "Errors found in script logs" >&2;
+	cat "$ROOT_DIR/stderr";
+	cat "$ROOT_DIR/stdout";
+	exit 1;
+fi
+
+USERS_JSON_FILE="$ROOT_DIR/opt/traffic_ops/install/data/json/users.json";
+USERS_TEST="
+import json
+import sys
+
+try:
+	with open('$USERS_JSON_FILE') as fd:
+		users_json = json.load(fd)
+except Exception as e:
+	print('Error loading users.json file:', e, file=sys.stderr)
+	exit(1)
+
+if not isinstance(users_json, dict) or len(users_json) != 2 or 'username' not in users_json or 'password' not in users_json:
+	print('Malformed users.json file - not an object or incorrect keys', file=sys.stderr)
+	exit(1)
+
+username = users_json['username']
+if not isinstance(username, str):
+	print('Username is not a string in users.json:', username, file=sys.stderr)
+	exit(1)
+
+if username != 'admin':
+	print('Incorrect username in users.json, expected: admin, got:', username, file=sys.stderr)
+	exit(1)
+
+password = users_json['password']
+if not isinstance(password, str):
+	print('Password is not a string in users.json:', password, file=sys.stderr)
+	exit(1)
+
+if not password.startswith('SCRYPT:16384:8:1:') or len(password.split(':')) != 6:
+	print('Malformed password field in users.json:', password, file=sys.stderr)
+	exit(1)
+
+exit(0)
+"
+
+/usr/bin/python3 -c "$USERS_TEST";
+
+POST_INSTALL_JSON="$ROOT_DIR/opt/traffic_ops/install/data/json/post_install.json";
+if [[ "$(cat $POST_INSTALL_JSON)" != "{}" ]]; then
+	echo "Incorrect post_install.json, expected: {}, got: $(cat $POST_INSTALL_JSON)" >&2;
+	exit 1;
+fi
+
+PROFILES_JSON_EXPECTED="{
+	\"tm.url\": \"https://localhost\",
+	\"cdn_name\": \"kabletown_cdn\",
+	\"dns_subdomain\": \"cdn1.kabletown.net\"
+}";
+
+PROFILES_JSON_ACTUAL="$(cat $ROOT_DIR/opt/traffic_ops/install/data/json/profiles.json)";
+if [[ "$PROFILES_JSON_ACTUAL" != "$PROFILES_JSON_EXPECTED" ]]; then
+	echo "Incorrect profiles.json, expected: $PROFILES_JSON_EXPECTED, got: $PROFILES_JSON_ACTUAL" >&2;
+	exit 1;
+fi
+
+DB_CONF_EXPECTED="production:
+    driver: postgres
+    open: host=localhost port=5432 user=traffic_ops password=twelve dbname=traffic_ops sslmode=disable";
+
+DB_CONF_ACTUAL="$(cat $ROOT_DIR/opt/traffic_ops/app/db/dbconf.yml)";
+if [[ "$DB_CONF_ACTUAL" != "$DB_CONF_EXPECTED" ]]; then
+	echo "Incorrect dbconf.yml, expected:" >&2;
+	echo "$DB_CONF_EXPECTED" >&2;
+	echo "got:" >&2;
+	echo "$DB_CONF_ACTUAL" >&2;
+	exit 1;
+fi
+
+CDN_CONF_TEST="
```

Comment:
Instead of storing this script to `CDN_CONF_TEST` and using it as a really long argument to `python -c`, if this script is a here document, `/usr/bin/python3` can be the command that the here document is standard input for.